### DISCUSSION TITLE
Log files for Wireguard?

### DISCUSSION THREAD

**Discussion by klack** [tenseiken/docker-qbittorrent-wireguard#5] [General]
Is there any way to access the log files for Wireguard?

**Comment by tenseiken** [comment]
Most of what I'm seeing is how to check server logs, and even then logging isn't enabled by default. It may be the same process for server and client, but it's not something I've tried before. I don't think the way I set things up uses the WireGuard kernel module, so you probably can't use dynamic debugging, but you may be able to use the iptables method found [here](https://www.procustodibus.com/blog/2021/03/wireguard-logs/). 

You'd need to modify the iptables.sh script and rebuild the image or you could modify it within your container by overwriting `/etc/qbittorrent/iptables.sh` with `docker cp` or overriding it with a volume. Either way, it looks like you'd want to change the iptables statements from line 103 to 113 and add the logging parameters mentioned in that link.

**Comment by klack** [comment]
Thanks for the reply!  I really like the Proton Port Forwarding capability.  I don't think I've seen it anywhere else.

My end goal is to use Grafana Loki for log aggregation and alerts.

The IPTables method gives some info, but I would really like to see this:
```
[Sat Mar  6 20:13:23 2021] wireguard: WireGuard 1.0.20201112 loaded. See www.wireguard.com for information.
[Sat Mar  6 20:13:23 2021] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[Sat Mar  6 20:35:33 2021] wireguard: wg0: Interface created
[Sat Mar  6 20:35:33 2021] wireguard: wg0: Peer 1 created
[Sat Mar  6 20:41:31 2021] wireguard: wg0: Invalid handshake initiation from 203.0.113.2:51820
```

Is there a way the log can be directed at syslog or journal within the container?  Promtail would be able to read and send logs this way.

**Comment by tenseiken** [comment]
I'm not really seeing any good options that work on Alpine. You can install syslog on Alpine through BusyBox but I don't see any way to direct logs from WireGuard to it. 

Looking at [this](https://www.wireguard.com/quickstart/#debug-info), it looks like in the userspace implementation, setting an environment variable LOG_LEVEL to "verbose" might do something. I see a little more detail in the docker logs when it runs wg-quick but I'm not seeing anything else.
